odbc/src/api/data_tests.rs

```diff
@@ -2798,194 +3193,138 @@ mod unit {
-                    MINKEY_COL,
-                    empty,
-                    SqlReturn::ERROR,
-                    "[MongoDB][API] BSON type minKey cannot be converted to ODBC type DateTime",
-                );
-                time_val_test(
-                    OID_COL,
-                    empty,
-                    SqlReturn::ERROR,
-                    "[MongoDB][API] BSON type objectId cannot be converted to ODBC type DateTime",
-                );
-                time_val_test(
-                    REGEX_COL,
-                    empty,
-                    SqlReturn::ERROR,
-                    "[MongoDB][API] BSON type regex cannot be converted to ODBC type DateTime",
-                );
-                time_val_test(
-                    STRING_COL,
-                    empty,
-                    SqlReturn::ERROR,
-                    "[MongoDB][API] invalid datetime format",
-                );
-            }
-
-            {
-                let mut null_val_test = |col: u16| {
-                    assert_eq!(
-                        SqlReturn::SUCCESS,
-                        SQLGetData(
-                            stmt_handle as *mut _,
-                            col,
-                            CDataType::SQL_C_TIME as i16,
-                            buffer,
-                            buffer_length,
-                            out_len_or_ind,
-                        )
-                    );
-                    assert_eq!(odbc_sys::NULL_DATA, *out_len_or_ind);
-                    assert_eq!(
-                        SqlReturn::NO_DATA,
-                        SQLGetData(
-                            stmt_handle as *mut _,
-                            col,
-                            CDataType::SQL_C_TIME as i16,
-                            buffer,
-                            buffer_length,
-                            out_len_or_ind,
-                        )
-                    );
-                };
 
-                null_val_test(NULL_COL);
-                null_val_test(UNDEFINED_COL);
-            }
-            let _ = Box::from_raw(buffer as *mut WChar);
+        unsafe {
+            assert_eq!(SqlReturn::ERROR, SQLFetch(stmt_handle as *mut _,));
+            assert_eq!(
+                format!("[MongoDB][API] No ResultSet"),
+                format!(
+                    "{}",
+                    (*stmt_handle)
+                        .as_statement()
+                        .unwrap()
+                        .errors
+                        .read()
+                        .unwrap()[0]
+                ),
+            );
             let _ = Box::from_raw(conn as *mut WChar);
             let _ = Box::from_raw(env as *mut WChar);
         }
     }
+
+    #[test]
+    fn sql_fetch_and_more_results_basic_functionality_test() {
+        sql_fetch_and_more_results_basic_functionality(SchemaMode::Standard);
+        sql_fetch_and_more_results_basic_functionality(SchemaMode::Simple);
+    }
```

Comment:
I did `cargo fmt` and everything got moved, so it looks like I made a lot of changes. All I really did though was make these new testing functions and add a parameter to the existing ones so that I could test both schema modes.